test_conformance/extensions/cl_khr_command_buffer/command_buffer_get_command_buffer_info.cpp

```diff
@@ -0,0 +1,334 @@
+        error = clGetCommandBufferInfoKHR(
+            command_buffer, CL_COMMAND_BUFFER_REFERENCE_COUNT_KHR,
+            sizeof(cl_uint), &new_ref_count, &ret_value_size);
+        test_error(error, "clGetCommandBufferInfoKHR failed");
+
+        test_expected_info(new_ref_count != init_ref_count);
+
+        return TEST_PASS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RunStateInfoTest()
+    {
+        cl_int error = CL_SUCCESS;
+
+        // lambda to verify given state
+        auto verify_state = [&](const cl_command_buffer_state_khr &expected) {
+            cl_command_buffer_state_khr state =
+                CL_COMMAND_BUFFER_STATE_INVALID_KHR;
+            size_t ret_value_size = 0;
+
+            cl_int error = clGetCommandBufferInfoKHR(
+                command_buffer, CL_COMMAND_BUFFER_STATE_KHR, sizeof(state),
+                &state, &ret_value_size);
+            test_error_ret(error, "clGetCommandBufferInfoKHR failed",
+                           TEST_FAIL);
+
+            test_expected_info(state != expected);
+
+            return TEST_PASS;
+        };
+
+        // verify recording state
+        error = verify_state(CL_COMMAND_BUFFER_STATE_RECORDING_KHR);
+        test_error(error, "verify_state failed");
+
+        // record command buffer
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+
+        // verify executable state
+        error = verify_state(CL_COMMAND_BUFFER_STATE_EXECUTABLE_KHR);
+        test_error(error, "verify_state failed");
+
+        error = clEnqueueFillBuffer(queue, out_mem, &pattern, sizeof(cl_int), 0,
+                                    data_size(), 0, nullptr, nullptr);
+        test_error(error, "clEnqueueFillBuffer failed");
+
+        clEventWrapper trigger_event = clCreateUserEvent(context, &error);
+        test_error(error, "clCreateUserEvent failed");
+
+        // enqueued command buffer blocked on user event
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 1,
+                                          &trigger_event, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        // verify pending state
+        error = verify_state(CL_COMMAND_BUFFER_STATE_PENDING_KHR);
+        test_error(error, "verify_state failed");
+
+        // execute command buffer
+        error = clSetUserEventStatus(trigger_event, CL_COMPLETE);
+        test_error(error, "clSetUserEventStatus failed");
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RunPropArrayInfoTest()
+    {
+        cl_int error = CL_SUCCESS;
+
+        // record command buffer
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+
+        size_t ret_value_size = 0;
+        cl_command_buffer_properties_khr combuf_props[16];
```

Comment:
Would prefer to make this a vector and then resize it, rather than a C array.
Something like:
```
std::vector<cl_command_buffer_properties_khr> combuf_props;
clGetCommandBufferInfoKHR(
            command_buffer, CL_COMMAND_BUFFER_PROPERTIES_ARRAY_KHR,
            0, NULL, &ret_value_size);
cl_uint num_props =  ret_value_size / sizeof(cl_command_buffer_properties_khr);
combuf_props.resize(num_props);
clGetCommandBufferInfoKHR(
            command_buffer, CL_COMMAND_BUFFER_PROPERTIES_ARRAY_KHR,
            ret_value_size, combuf_props.data(), NULL );

Reply:
Done